sdk/resourcemanager/Azure.ResourceManager/src/Resources/Custom/PolicyAssignmentData.cs

```diff
@@ -57,8 +57,19 @@ internal PolicyAssignmentData(ResourceIdentifier id, string name, ResourceType r
 
         /// <summary> The location of the policy assignment. Only required when utilizing managed identity. </summary>
         public AzureLocation? Location { get; set; }
+
+        private SystemAssignedServiceIdentity _identity;
         /// <summary> The managed identity associated with the policy assignment. </summary>
-        public SystemAssignedServiceIdentity Identity { get; set; }
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public SystemAssignedServiceIdentity Identity
+        {
+            get => _identity ??= ManagedIdentity == null ? null : new SystemAssignedServiceIdentity(ManagedIdentity.PrincipalId, ManagedIdentity.TenantId, ManagedIdentity.ManagedServiceIdentityType.ToString());
```

Comment:
I feel like we need some rigorous test cases around these conversions to make sure there are no errors and they persist.

If we are making a new one every time wouldn't I lose my changes after I did a set?  Also what happens if ManagedIdentity is null?

Lets add the test cases to answer these questions.

Reply:
With more tests, I found it very difficult to always keep Identity and ManagedIdentity data in sync with the way to update identity with `data.Identity.SystemAssignedServiceIdentityType = ...` as it's beyond the reach of setters in PolicyAssignmentData. Therefore, I updated the code to assume a user will only either use Identity or ManagedIdentity which I think is a reasonable assumption and added tests for different scenarios.